app/main.py

```diff
@@ -2,0 +1,31 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        print(f"Animal __init__ called with: {name}, health: {health}")
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+# Use
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
```

Comment:
Code Style: Use interpolation instead of concatenation. It's a good practice to use string interpolation as it's more readable and efficient.